finesse/gui/temperature_plot.py

```diff
@@ -81,7 +81,7 @@ def _create_figure(self) -> None:
         hot_bb_temp: list[float | None] = [None] * self._figure_num_pts
         cold_bb_temp: list[float | None] = [None] * self._figure_num_pts
 
-        t[0] = datetime.now().timestamp()
+        t[0] = None
         hot_bb_temp[0] = 25.0
         cold_bb_temp[0] = 25.0
 
```

Comment:
I *think* these lines can just be deleted altogether:

```suggestion
```

@dc2917 is there a reason we need an initial temperature set here? It seems a little odd. I've tried deleting these lines locally and things still seem to work.

Reply:
Yeah, I think this used to be required but the `pop` and `append` in `_update_figure` presumably now take care of whatever the problem was.

Reply:
It would be nice if a Windows-user could check they don't have any issues though (i.e. as in https://github.com/ImperialCollegeLondon/FINESSE/issues/220).

Reply:
I guess the answer is "it does not work on Windows". I get the following when trying to launch FINESSE:

```
Traceback (most recent call last):
  File "<frozen runpy>", line 198, in _run_module_as_main
  File "<frozen runpy>", line 88, in _run_code
  File "C:\Users\dalonsoa\Projects\FINESSE\finesse\__main__.py", line 6, in <module>
    run()
  File "C:\Users\dalonsoa\Projects\FINESSE\finesse\__init__.py", line 24, in run
    window = MainWindow()
             ^^^^^^^^^^^^
  File "C:\Users\dalonsoa\Projects\FINESSE\finesse\gui\main_window.py", line 89, in __init__
    bb_monitor: QGroupBox = TemperaturePlot()
                            ^^^^^^^^^^^^^^^^^
  File "C:\Users\dalonsoa\Projects\FINESSE\finesse\gui\temperature_plot.py", line 34, in __init__
    layout = self._create_controls()
             ^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\dalonsoa\Projects\FINESSE\finesse\gui\temperature_plot.py", line 62, in _create_controls
    self._create_figure()
  File "C:\Users\dalonsoa\Projects\FINESSE\finesse\gui\temperature_plot.py", line 104, in _create_figure
    self._canvas.draw()
  File "C:\Users\dalonsoa\AppData\Local\pypoetry\Cache\virtualenvs\finesse-VOhfgRFV-py3.11\Lib\site-packages\matplotlib\backends\backend_agg.py", line 388, in draw
    self.figure.draw(self.renderer)
  File "C:\Users\dalonsoa\AppData\Local\pypoetry\Cache\virtualenvs\finesse-VOhfgRFV-py3.11\Lib\site-packages\matplotlib\artist.py", line 95, in draw_wrapper
    result = draw(artist, renderer, *args, **kwargs)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\dalonsoa\AppData\Local\pypoetry\Cache\virtualenvs\finesse-VOhfgRFV-py3.11\Lib\site-packages\matplotlib\artist.py", line 72, in draw_wrapper
    return draw(artist, renderer)
           ^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\dalonsoa\AppData\Local\pypoetry\Cache\virtualenvs\finesse-VOhfgRFV-py3.11\Lib\site-packages\matplotlib\figure.py", line 3148, in draw
    self.get_layout_engine().execute(self)
  File "C:\Users\dalonsoa\AppData\Local\pypoetry\Cache\virtualenvs\finesse-VOhfgRFV-py3.11\Lib\site-packages\matplotlib\layout_engine.py", line 273, in execute
    return do_constrained_layout(fig, w_pad=w_pad, h_pad=h_pad,
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\dalonsoa\AppData\Local\pypoetry\Cache\virtualenvs\finesse-VOhfgRFV-py3.11\Lib\site-packages\matplotlib\_constrained_layout.py", line 116, in do_constrained_layout
    make_layout_margins(layoutgrids, fig, renderer, h_pad=h_pad,
  File "C:\Users\dalonsoa\AppData\Local\pypoetry\Cache\virtualenvs\finesse-VOhfgRFV-py3.11\Lib\site-packages\matplotlib\_constrained_layout.py", line 381, in make_layout_margins
    pos, bbox = get_pos_and_bbox(ax, renderer)
                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\dalonsoa\AppData\Local\pypoetry\Cache\virtualenvs\finesse-VOhfgRFV-py3.11\Lib\site-packages\matplotlib\_constrained_layout.py", line 634, in get_pos_and_bbox
    tightbbox = martist._get_tightbbox_for_layout_only(ax, renderer)
                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\dalonsoa\AppData\Local\pypoetry\Cache\virtualenvs\finesse-VOhfgRFV-py3.11\Lib\site-packages\matplotlib\artist.py", line 1411, in _get_tightbbox_for_layout_only
    return obj.get_tightbbox(*args, **{**kwargs, "for_layout_only": True})
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\dalonsoa\AppData\Local\pypoetry\Cache\virtualenvs\finesse-VOhfgRFV-py3.11\Lib\site-packages\matplotlib\_api\deprecation.py", line 454, in wrapper
    return func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\dalonsoa\AppData\Local\pypoetry\Cache\virtualenvs\finesse-VOhfgRFV-py3.11\Lib\site-packages\matplotlib\axes\_base.py", line 4395, in get_tightbbox
    ba = martist._get_tightbbox_for_layout_only(axis, renderer)
         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\dalonsoa\AppData\Local\pypoetry\Cache\virtualenvs\finesse-VOhfgRFV-py3.11\Lib\site-packages\matplotlib\artist.py", line 1411, in _get_tightbbox_for_layout_only
    return obj.get_tightbbox(*args, **{**kwargs, "for_layout_only": True})
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\dalonsoa\AppData\Local\pypoetry\Cache\virtualenvs\finesse-VOhfgRFV-py3.11\Lib\site-packages\matplotlib\axis.py", line 1334, in get_tightbbox
    ticks_to_draw = self._update_ticks()
                    ^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\dalonsoa\AppData\Local\pypoetry\Cache\virtualenvs\finesse-VOhfgRFV-py3.11\Lib\site-packages\matplotlib\axis.py", line 1276, in _update_ticks
    major_labels = self.major.formatter.format_ticks(major_locs)
                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\dalonsoa\AppData\Local\pypoetry\Cache\virtualenvs\finesse-VOhfgRFV-py3.11\Lib\site-packages\matplotlib\ticker.py", line 216, in format_ticks
    return [self(value, i) for i, value in enumerate(values)]
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\dalonsoa\AppData\Local\pypoetry\Cache\virtualenvs\finesse-VOhfgRFV-py3.11\Lib\site-packages\matplotlib\ticker.py", line 216, in <listcomp>
    return [self(value, i) for i, value in enumerate(values)]
            ^^^^^^^^^^^^^^
  File "C:\Users\dalonsoa\AppData\Local\pypoetry\Cache\virtualenvs\finesse-VOhfgRFV-py3.11\Lib\site-packages\matplotlib\ticker.py", line 323, in __call__
    return self.func(x, pos)
           ^^^^^^^^^^^^^^^^^
  File "C:\Users\dalonsoa\Projects\FINESSE\finesse\gui\temperature_plot.py", line 160, in _xtick_format_fcn
    return datetime.fromtimestamp(val).strftime("%H:%M:%S")
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^
OSError: [Errno 22] Invalid argument

```

Reply:
So yeah, issue #220 then. 

As I mentioned in that issue, we can set t[0] to whatever we want, as long as it doesn't cause any of the x tick values to be less than 0. I just chose `datetime.now()` at the time, but we could just do midday or something.

Reply:
Good catch @dc2917! I don't think midday would work if you were recording *before*
midday though... Did you mean midnight?

Reply:
It only causes a problem if the argument to `datetime.fromtimestamp()` is a negative float. `_xtick_format_fcn` turns a (float) timestamp into a string via `datetime.fromtimestamp(val).strftime("%H:%M:%S")`.

If you create a plot with empty x and y data, matplotlib centres the plot on (0,0). So if we don't set `t[0]`, the x axis will be centred on 0, so x ticks left of centre will be negative (e.g. `datetime.fromtimestamp(-0.1).strftime("%H:%M:%S")`), raising the error.

Midday any day on or after 1st Jan 1970 would work. Midnight any day after (but not on) 1st Jan 1970 would work. `datetime.now()` seems the most logical to me, but it could literally be any timestamp after 00:03:00 on 1st Jan 1970 (matplotlib creates x-ticks at 00:00:00, 00:01:00, 00:02:00, 00:03:00, ..., 00:06:00 if `t[0]` corresponds to 00:03:00, so anything before 00:03:00 would make the first x tick before 00:00:00 on 1st Jan 1970, i.e. a negative float).

The value is never recorded anywhere, it's just for matplotlib to be able to initialise the figure. Once the connection to the temperature monitors takes place, `_update_figure` is called and these initial x and y values are `pop`ped from their respective lists.

Reply:
I seeeeeee... Thanks for the explanation!

Would it work to do:

```py
        return datetime.fromtimestamp(max(val, 0)).strftime("%H:%M:%S")
```

Reply:
Yeah, I guess that would just set all the ticks strings to 00:00:00, and we could remove setting the initial values.

Reply:
Okay, so I'll change the return value of `_xtick_format_fcn` to `datetime.fromtimestamp(max(val, 0)).strftime("%H:%M:%S")` and remove the initialisation in `_create_figure` and I think we're good.